### PR TITLE
Spanner sample

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
Added the autoconfig earlier today, so here is the Spanner sample added with README stuff!

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+spring.cloud.gcp.spanner.projectId=amp-demo-landing-page
```
Comment (balopat) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/resources/application.properties]: let's change this to `<your-project-id>`

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner/pom.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			<artifactId>spring-cloud-gcp-starter</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-gcp-data-spanner</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-tx</artifactId>
+		</dependency>
```
Comment (balopat) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner/pom.xml]: why spring-tx?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner/pom.xml]: I'm not sure specifically what needs it inside the spring boot rest application section of the app, but without it I get a class-not-found exception at startup.

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/resources/application.properties (hunk 1 above)
Comment (ChengyuanZhao) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/resources/application.properties]: hahaha oh yes.

Comment (ChengyuanZhao) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/resources/application.properties]: @balopat changes made!

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner/pom.xml (hunk 2 above)
Comment (balopat) [reply spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner/pom.xml]: I see, [TransactionalProxy](https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/repository/core/support/RepositoryFactorySupport.java#L303) is used by the `RepositoryFactorySupport` class

**Comment by balopat** [comment]
LGTM.

**Pull request merged**